roseau/load_flow/models/lines/lines.py

```diff
@@ -217,10 +218,15 @@ def __init__(
         super().__init__(id, bus1, bus2, phases1=phases, phases2=phases, geometry=geometry, **kwargs)
         self.phases = phases
         self.ground = ground
-        self.length = length
+        self._length = length
         self.parameters = parameters
         self._initialized = True
 
+    @property
+    @ureg.wraps("km", (None,), strict=False)
+    def length(self) -> Q_:
+        return self._length
+
```

Comment:
I think you'll need a setter as well? If you add it you may want to invalidate the results in the setter if the length changes